kolibri/core/logger/api.py

```diff
@@ -675,6 +675,36 @@ def _create_attempt(
             **interaction
         )
 
+    def _get_attemptlog(self, session_id, masterylog_id, user, interaction):
+        if "id" in interaction:
+            try:
+                return AttemptLog.objects.get(
+                    id=interaction["id"],
+                    masterylog_id=masterylog_id,
+                    user=user,
+                )
+            except AttemptLog.DoesNotExist:
+                raise ValidationError("Invalid attemptlog id specified")
+        elif interaction.get("replace", False):
+            try:
+                if user:
+                    return AttemptLog.objects.get(
+                        masterylog_id=masterylog_id,
+                        user=user,
+                        item=interaction["item"],
+                    )
+                else:
+                    # If this is an anonymous user, then the best we can do is
+                    # try to update any previous attempt from this session.
+                    return AttemptLog.objects.get(
+                        masterylog_id=masterylog_id,
+                        sessionlog_id=session_id,
+                        user=user,
```

Comment:
Even though I guess it's equivalent, it might be clearer to put `user=None` here, if I'm reading correctly?

Reply:
Yes, and `masterylog_id` will also be `None`.